src/protocol/sort/generate_permutation_opt.rs

```diff
@@ -243,11 +243,52 @@ mod tests {
             v2.validate(result2),
         )
         .await;
-        let mut mpc_sorted_list = (0..u128::try_from(COUNT).unwrap()).collect::<Vec<_>>();
+        let mut mpc_sorted_list = (0..u128::try_from(count).unwrap()).collect::<Vec<_>>();
         for (match_key, index) in zip(match_keys, result.reconstruct()) {
             mpc_sorted_list[index.as_u128() as usize] = match_key.as_u128();
         }
 
         assert_eq!(expected, mpc_sorted_list);
     }
+
+    #[tokio::test]
+    pub async fn malicious() {
+        sortn(10).await;
+    }
+
+    /// Passing 32 records for Fp31 doesn't work.
+    #[tokio::test]
+    #[should_panic]
+    async fn fp31_overflow() {
```

Comment:
I love Fp31 field. Do you want to check the panic message too? I've seen cases when tests continue to panic but for a different reason :)